### ISSUE TITLE
Does not reach first iteration of training. (Only edit pose_config.yaml manually if you do so consistently)

### ISSUE THREAD

**Issue by elliottabe** [DeepLabCut/DeepLabCut#43]
**Describe the problem**
When training on labeled data, the code seems to get stuck after establishing the connection with the GPU.  This only happens occasionally, since I was able to train on a different data set.  When i look at the nvidia-smi the GPU does not appear to be processing anything even though the memory is reserved.  

**To Reproduce**
1. Follow the Generating training data steps to create and begin training. 
2. edit some parts but not others. 

**Screenshots**
log.txt file when program gets stuck. 

2018-08-14 23:47:21 Config:
{'all_joints': [[0], [1], [2], [3], [4], [5], [6]],
 'all_joints_names': ['P1', 'P2', 'P3', 'P4', 'P5', 'P6', 'P7'],
 'batch_size': 1,
 'crop': False,
 'crop_pad': 0,
 'dataset': '../../UnaugmentedDataSet_FishTailAug14/FishTail_Elliott95shuffle1.mat',
 'dataset_type': 'default',
 'display_iters': 1000,
 'fg_fraction': 0.25,
 'global_scale': 0.8,
 'init_weights': '../../pretrained/resnet_v1_50.ckpt',
 'intermediate_supervision': False,
 'intermediate_supervision_layer': 12,
 'location_refinement': True,
 'locref_huber_loss': True,
 'locref_loss_weight': 0.05,
 'locref_stdev': 7.2801,
 'log_dir': 'log',
 'max_input_size': 1000,
 'mean_pixel': [123.68, 116.779, 103.939],
 'mirror': False,
 'multi_step': [[0.005, 10000],
                [0.02, 430000],
                [0.002, 730000],
                [0.001, 1030000]],
 'net_type': 'resnet_50',
 'num_joints': 6,
 'optimizer': 'sgd',
 'pos_dist_thresh': 17,
 'regularize': False,
 'save_iters': 50000,
 'scale_jitter_lo': 0.5,
 'scale_jitter_up': 1.5,
 'scoremap_dir': 'test',
 'shuffle': True,
 'snapshot_prefix': './snapshot',
 'stride': 8.0,
 'use_gt_segm': False,
 'video': False,
 'video_batch': False,
 'weigh_negatives': False,
 'weigh_only_present_joints': False,
 'weigh_part_predictions': False,
 'weight_decay': 0.0001}
2018-08-14 23:47:21 From /home/eabe/DeepLabCut/pose-tensorflow/nnet/pose_net.py:52: calling resnet_arg_scope (from tensorflow.contrib.slim.python.slim.nets.resnet_utils) with is_training is deprecated and will be removed after 2017-08-01.
Instructions for updating:
Pass is_training directly to the network instead of the arg_scope.
2018-08-14 23:47:23 logits.dtype=<dtype: 'float32'>.
2018-08-14 23:47:23 multi_class_labels.dtype=<dtype: 'float32'>.
2018-08-14 23:47:23 losses.dtype=<dtype: 'float32'>.
2018-08-14 23:55:32 Restoring parameters from ../../pretrained/resnet_v1_50.ckpt

**Additional context**
Training on 77 images, I have tried with 3 points and 7 points and still gets stuck.  It seems like it is getting stuck when loading in the .mat file.  When switching out the path to another one and changing to the appropriate number of joints, it works.  I have tried recreating the training set and still does not get past the initial set up.

**Comment by MMathisLab** [comment]
seems the naming and number of joints is inconsistent. The label is repeated “P6,” so it thinks it has 6 points, which you can see at ‘num_joints’ : 6... try changing the names to be independent variables and re-create the training files.

**Comment by elliottabe** [comment]
@MMathisLab I tried correcting the labels and recreated the files but the network gets stuck at the same place.

**Comment by AlexEMG** [comment]
Did you recreate the ```pose_config.yaml``` file following all the steps? I'm not sure this is a DLC bug - given that other configuration files work (even for you). Perhaps it would be interesting to find out what exactly goes wrong for this example (by debugging it).

Your file is inconsistent, due to:

'num_joints': 6,
'all_joints': [[0], [1], [2], [3], [4], [5], [6]],
'all_joints_names': ['P1', 'P2', 'P3', 'P4', 'P5', 'P6', 'P6'], 

You just updated the latter to: 'all_joints_names': ['P1', 'P2', 'P3', 'P4', 'P5', 'P6', 'P7'], but num_joints is still 6. If you used the toolbox, these entries are generated by 

            "num_joints": len(bodyparts),
            "all_joints": [[i] for i in range(len(bodyparts))],
            "all_joints_names": bodyparts

in
```https://github.com/AlexEMG/DeepLabCut/blob/master/Generating_a_Training_Set/Step4_GenerateTrainingFileFromLabelledData.py```

So the **length item created by the code would be 7**.

Thus, please either use the code as provided, or try to change things consistently.

**Comment by MMathisLab** [comment]
(Also when you recreated the file, you need to delete the h5 file, it will not overwrite itself!) [that is in Step2 of making a training set]

**Comment by triccelli** [comment]
@eabeUO were you able to fix this issue? If so how? I am having the same issue, but num_joints, all_joints, and all_joints_names are consistent in log

**Comment by elliottabe** [comment]
@triccelli I was not able to understand what fixed the issue, but I regenerated the training set from scratch (deleting the folders and selecting new points) and the network began to train.

**Comment by triccelli** [comment]
@eabeUO  Okay thank you! And everything else was exactly the same as you had done before when you regenerated the training set as far as you are aware? I will try that
